intro.js

```diff
@@ -1370,6 +1371,27 @@
   }
 
   /**
+  * Iterates arrays
+  *
+  * @param {Array} arr
+  * @param {Function} forEachFnc
+  * @param {Function} completeFnc
+  * @return {Null}
+  */
+  function _forEach(arr, forEachFnc, completeFnc) {
+    // in case arr is an empty query selector node list
+    if (arr) {
+      for (var i = 0, len = arr.length; i < len; i++) {
+        forEachFnc(arr[i], i);
```

Comment:
what about `forEachFnc.call(this, arr[i], i);`? just to pass Introjs instance to the fn. Same for completeFnc.

Reply:
Some of them don't need it, and because it's a private function, this might be ambiguous and we control all the calls, so we can use `.bind(this)` where appropriate.  Right now most of them look like this:

```javascript
_forEach(arr, function (item) {
  // do something here within the same context
}.bind(this));
```

Reply:
ah I see.